<launch>
  <arg name="model" default="burger"/>
  <arg name="slam_methods" default="gmapping" doc="slam type [gmapping, hector]"/>
 <!----> <arg name="use_rviz" default="false" doc="set to true to enable RViz"/>

  <!-- Node xuất bản transform từ map đến odom -->
  <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0.0 0.0 0.0 0.0 0.0 0.0 map odom 100" />
  
  <!-- SLAM: Gmapping, Hector -->
  <include file="$(find turtlebot3_slam)/launch/turtlebot3_$(arg slam_methods).launch">
    <arg name="model" value="$(arg model)"/>
  </include>

  <!-- RVIZ - chỉ khởi động khi được yêu cầu -->
  <group if="$(arg use_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find turtlebot3_slam)/rviz/turtlebot3_$(arg slam_methods).rviz"/>
  </group>
</launch>